### PR TITLE
refactor(server_media)_: pass a func to get the images instead of using the cache

### PR DESCRIPTION
**Pull request by jrainville**
Issue raised in this PR https://github.com/status-im/status-go/pull/6118#discussion_r1857255843

The community cache that the image server was using was not intended to be used for that. It can be invalidated at any moment. Also, it did not contain changes made by admins (admin events).
Using this new approach, we pass functions from the community manager to the media server so that it can have access to the correct community description.

### CODE REVIEW

**Review by osmaczko — APPROVED** [review]
:+1:

File: server/server_media.go [hunk 1]
```diff
@@ -25,10 +27,12 @@ func WithMediaServerDisableTLS(disableTLS bool) MediaServerOption {
...
-	db              *sql.DB
-	downloader      *ipfs.Downloader
-	multiaccountsDB *multiaccounts.Database
-	walletDB        *sql.DB
+	db                       *sql.DB
+	downloader               *ipfs.Downloader
+	multiaccountsDB          *multiaccounts.Database
+	walletDB                 *sql.DB
+	getCommunityImagesReader func(communityID string) (map[string]*protobuf.IdentityImage, error)
+	getCommunityTokenReader  func(communityID string) ([]*protobuf.CommunityTokenMetadata, error)
```
Comment (igor-sirotin) [inline server/server_media.go]: I guess it's either `get...` or `...Reader`, not both? 🤔

**Review by igor-sirotin — APPROVED** [review]
I looked at this whole relationship between Communities and MediaServer, they inter-use each other, which is not quite good. `MediaServer` needs `CommunityManager` for the things as in this PR. And vice versa for `Community.MarshalJSON` here:
https://github.com/status-im/status-go/blob/72f7ed7ced4d7a0838874fe04c00eec0290eae86/protocol/communities/community.go#L267

Your solution here is good now, I don't see any quick and simple way to change what we've got. But would be nice to improve this at some point in future.

**Review by qfrank — APPROVED** [review]
@jrainville Do you think if it's a good idea to add a referrence comment(including @osmaczko explanation) on `preprocessDescription` ? 🤔

**Comment by jrainville** [comment]
@igor-sirotin there are two tests failing in `BlockchainHealthManagerSuite`, but I saw that they are flaky on develop too.
What's the procedure for that?
I'll restart the tests for now

**Comment by osmaczko** [comment]
> @igor-sirotin there are two tests failing in `BlockchainHealthManagerSuite`, but I saw that they are flaky on develop too. What's the procedure for that? I'll restart the tests for now

`TestSerializedCommunities ` is failing now

**Pull request merged**